caffe2/utils/CMakeLists.txt

```diff
@@ -1,47 +1,95 @@
-# ---[ GPU files
-# ------[ cuDNN
-file(GLOB_RECURSE tmp *_cudnn.cc)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# ------[ general GPU
-file(GLOB_RECURSE tmp *_gpu.cc)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# ------[ CUDA sources
-file(GLOB_RECURSE tmp *.cu)
-set(Caffe2_GPU_SRCS ${Caffe2_GPU_SRCS} ${tmp})
-# exclude test files
-file(GLOB_RECURSE tmp *_test.cc)
-exclude(Caffe2_GPU_SRCS "${Caffe2_GPU_SRCS}" ${tmp})
-
-# ---[ CPU files.
-file(GLOB_RECURSE tmp *.cc)
-set(Caffe2_CPU_SRCS ${Caffe2_CPU_SRCS} ${tmp})
-# exclude test files and gpu files
-file(GLOB_RECURSE tmp *_test.cc)
-exclude(Caffe2_CPU_SRCS "${Caffe2_CPU_SRCS}" ${tmp})
-exclude(Caffe2_CPU_SRCS "${Caffe2_CPU_SRCS}" ${Caffe2_GPU_SRCS})
+project(c10_utils CXX C)
```

Comment:
Short term: This allows me to enforce higher code quality standards for new source files (i.e. more warnings flags) without having to change existing code all at once.

Long term: I think we should move towards a more modularized build setup throughout the whole code base, see also the quip document I wrote about this.

Reply:
Looks good - I saw you got rid of the project() line. Basically, I am mainly concerned about the complexity that project() introduces to the codebase. I do not have problem with the add_library(... object) settings.

By the way, it might be possible to do a codemod to improve the codebase's -W settings in general, but that is not a blocking item.